sgkit_vcf/vcf_reader.py

```diff
@@ -0,0 +1,299 @@
+                    call_genotype_phased=dict(chunks=(chunk_length, chunk_width)),
+                    variant_allele=dict(chunks=(chunk_length, n_allele)),
+                    variant_contig=dict(chunks=(chunk_length,)),
+                    variant_id=dict(chunks=(chunk_length,)),
+                    variant_id_mask=dict(chunks=(chunk_length,)),
+                    variant_position=dict(chunks=(chunk_length,)),
+                    sample_id=dict(chunks=(chunk_width,)),
+                )
+
+                ds.to_zarr(output, mode="w", encoding=encoding)
+                first_variants_chunk = False
+            else:
+                # Append along the variants dimension
+                ds.to_zarr(output, append_dim=DIM_VARIANT)
+
+
+def vcf_to_zarr_parallel(
+    input: Union[PathType, Sequence[PathType]],
+    output: PathType,
+    regions: Union[None, Sequence[str], Sequence[Optional[Sequence[str]]]],
+    chunk_length: int = 10_000,
+    chunk_width: int = 1_000,
+    tempdir: Optional[Path] = None,
+) -> None:
+    """Convert specified regions of one or more VCF files to zarr files, then concat, rechunk, write to zarr"""
+
+    if tempdir is None:
+        tempdir = Path(tempfile.mkdtemp(prefix="vcf_to_zarr_"))
+
+    paths = vcf_to_zarrs(input, tempdir, regions, chunk_length, chunk_width)
+
+    ds = zarrs_to_dataset(paths, chunk_length, chunk_width)
+
+    # Ensure Dask task graph is efficient, see https://github.com/dask/dask/issues/5105
+    with dask.config.set({"optimization.fuse.ave-width": 50}):
+        ds.to_zarr(output, mode="w")
+
+    # Delete intermediate files from temporary directory
+    for path in paths:
+        shutil.rmtree(path)
+
+
+def vcf_to_zarrs(
+    input: Union[PathType, Sequence[PathType]],
+    output: PathType,
+    regions: Union[None, Sequence[str], Sequence[Optional[Sequence[str]]]],
+    chunk_length: int = 10_000,
+    chunk_width: int = 1_000,
+) -> Sequence[Path]:
+    """Convert specified regions of one or more VCF files to zarr files."""
+
+    if isinstance(input, str) or isinstance(input, Path):
+        # Single input
+        inputs: Sequence[PathType] = [input]
+        assert regions is not None  # this would just be sequential case
+        input_regions: Sequence[Optional[Sequence[str]]] = [regions]  # type: ignore
+    else:
+        # Multiple inputs
+        inputs = input
+        if regions is None:
+            input_regions = [None] * len(inputs)
+        else:
+            if len(regions) == 0 or isinstance(regions[0], str):
+                raise ValueError(
+                    f"For multiple inputs, multiple input regions must be a sequence of sequence of strings: {regions}"
+                )
+            input_regions = regions
+
+    assert len(inputs) == len(input_regions)
+
+    tasks = []
+    parts = []
+    for i, input in enumerate(inputs):
+        filename = Path(input).name
+        input_region_list = input_regions[i] or [None]  # type: ignore
```

Comment:
I'm baffled here - what does this do?

Reply:
The multi-region handling is a bit tricky - I wonder if there's a better way of structuring it (in terms of types). I have added a comment in the meantime.

https://github.com/pystatgen/sgkit-vcf/pull/1/commits/149ab37e91227196e479dbc86297b35b608d4e85